mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/processing/impl/VectorFieldAnnotationProcessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.impl;
+
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.VectorField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessor;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessorContext;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStep;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingVectorFieldStep;
+
+public class VectorFieldAnnotationProcessor implements PropertyMappingAnnotationProcessor<VectorField> {
+
+	@Override
+	public final void process(PropertyMappingStep mappingContext, VectorField annotation,
+			PropertyMappingAnnotationProcessorContext context) {
+		String cleanedUpRelativeFieldName = context.toNullIfDefault( annotation.name(), "" );
+
+		PropertyMappingVectorFieldStep fieldContext = mappingContext.vectorField( cleanedUpRelativeFieldName )
+				.beamWidth( annotation.beamWidth() )
+				.maxConnections( annotation.maxConnections() )
```

Comment:
You should probably handle defaults here.
See what we do with the decimal scale in `ScaledNumberFieldProcessor`.

Reply:
And actually you did handle defaults for other fields below...